store/sqlstore/channel_store.go

```diff
@@ -949,17 +949,37 @@ func (s SqlChannelStore) GetChannels(teamId string, userId string, includeDelete
 	})
 }
 
-func (s SqlChannelStore) GetAllChannels(offset int, limit int, includeDeleted bool) store.StoreChannel {
+func (s SqlChannelStore) GetAllChannels(offset int, limit int, opts model.ChannelSearchOpts) store.StoreChannel {
```

Comment:
I'm going to create a separate struct that does not include the `ExcludeDefaultChannels` field so that it's clear that that field doesn't function at the data-access tier.